core/src/main/java/org/elasticsearch/common/lucene/search/function/FiltersFunctionScoreQuery.java

```diff
@@ -55,53 +54,63 @@ public FilterFunction(Query filter, ScoreFunction function) {
 
         @Override
         public boolean equals(Object o) {
-            if (this == o)
+            if (this == o) {
                 return true;
-            if (o == null || getClass() != o.getClass())
+            }
+            if (o == null || getClass() != o.getClass()) {
                 return false;
-
+            }
             FilterFunction that = (FilterFunction) o;
-
-            if (filter != null ? !filter.equals(that.filter) : that.filter != null)
-                return false;
-            if (function != null ? !function.equals(that.function) : that.function != null)
-                return false;
-
-            return true;
+            return Objects.equals(this.filter, that.filter) && Objects.equals(this.function, that.function);
         }
 
         @Override
         public int hashCode() {
-            int result = filter != null ? filter.hashCode() : 0;
-            result = 31 * result + (function != null ? function.hashCode() : 0);
-            return result;
+            return Objects.hash(super.hashCode(), filter, function);
         }
     }
 
-    public static enum ScoreMode {
-        First, Avg, Max, Sum, Min, Multiply
+    public enum ScoreMode implements Writeable<ScoreMode> {
+        FIRST, AVG, MAX, SUM, MIN, MULTIPLY;
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeVInt(this.ordinal());
+        }
+
+        @Override
+        public ScoreMode readFrom(StreamInput in) throws IOException {
+            int ordinal = in.readVInt();
+            if (ordinal < 0 || ordinal >= values().length) {
+                throw new IOException("Unknown ScoreMode ordinal [" + ordinal + "]");
+            }
+            return values()[ordinal];
+        }
+
+        public static ScoreMode readScoreModeFrom(StreamInput in) throws IOException {
+            return ScoreMode.MULTIPLY.readFrom(in);
+        }
+
+        public static ScoreMode fromString(String scoreMode) {
+            return valueOf(scoreMode.toUpperCase(Locale.ROOT));
+        }
     }
 
     Query subQuery;
     final FilterFunction[] filterFunctions;
     final ScoreMode scoreMode;
     final float maxBoost;
-    private Float minScore;
+    private final Float minScore;
```

Comment:
tried that, it isn't straight-forward. `null` seems to have a meaning here: https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/common/lucene/search/function/CustomBoostFactorScorer.java#L39